src/main/java/edu/ucsb/cs156/happiercows/entities/Commons.java

```diff
@@ -29,6 +29,7 @@
   private double milkPrice;
   private double startingBalance;
   private LocalDateTime startingDate;
+  private LocalDateTime lastDate;
```

Comment:
You have the fields here.  You do not need them, and should not store them redundantly on the UserCommons.